src/storage/catalog_class.c

```diff
@@ -2906,12 +2906,8 @@ catcls_expand_or_value_by_subset (THREAD_ENTRY * thread_p, OR_VALUE * value_p)
       if (DB_VALUE_TYPE (&element) == DB_TYPE_OID)
 	{
 	  oid_p = DB_PULL_OID (&element);
-	  /* 
-	   * Need to get class OID using SNAPSHOT_TYPE_NONE, since
-	   * the OID may be referred from other catalog class and we
-	   * want the class OID of referred OID.
-	   */
-	  scan_code = heap_get_class_oid_with_lock (thread_p, &class_oid, oid_p, SNAPSHOT_TYPE_NONE, NULL_LOCK);
+
+	  scan_code = heap_get_class_oid (thread_p, oid_p, &class_oid);
```

Comment:
Again, I am not sure heap_get_class_oid is correct here. Just below, the code verifies if object is deleted, clearing the error? I am not about the impact of your change :).